pysatMadrigal/instruments/methods/general.py

```diff
@@ -533,9 +985,18 @@ def get_remote_filenames(inst_code=None, kindat=None, user=None, password=None,
                 date_array))
         start = date_array.min()
         stop = date_array.max()
+
+    # If NoneType was supplied for start or stop, set to defaults
+    if start is None:
+        start = dt.datetime(1900, 1, 1)
+
+    if stop is None:
+        stop = dt.datetime.utcnow()
+
     # If start and stop are identical, increment
     if start == stop:
         stop += dt.timedelta(days=1)
+
```

Comment:
```suggestion
    istr = "Found {:d} Madrigal experiments between {:s} and {:s}".format(
        len(exp_list), start.strftime('%d %B %Y'), stop.strftime('%d %B %Y'))
    logger.info(istr)
```

This is something I added a few lines down but this is the closest I can get via GitHub.

Reply:
For some reason when I run your example what is actually downloaded is data for Jan 1, 2015.
```
pysat INFO: Connecting to Madrigal
pysat INFO: Connection established.
pysat INFO: Found 1 Madrigal experiments between 30 December 2015 and 31 December 2015
pysat INFO: /Volumes/BlackBox/ScienceData/madrigal/pandas/180/dmp20150101.002.hdf5 already exists. Skipping.
```

```
In [7]: dmsp_abi.files.files
Out[7]: 
2015-01-01    dmp20150101.002.hdf5
dtype: object
```

Reply:
The output from list_remote_files leads to the Jan 1 date, 
```
/opt/cedar3/experiments/2015/dmp/01jan15/dmp20150101.002.hdf5
```

Reply:
I put this in as well and get the same thing:
```
            file_list = web_data.getExperimentFiles(exp.id)
            for f in file_list:
                print('In loop file ', f.name)
```
```
In loop file  /opt/cedar3/experiments/2015/dmp/01jan15/dmp20150101.002.hdf5
```

Reply:
I am not sure what you're asking for.  Can you make the change to the branch and commit it?

Reply:
> I am not sure what you're asking for. Can you make the change to the branch and commit it?

Committed the update to feedback.


What I was also probably unclear about is `dmsp_abi` is downloading the wrong day of data, it is grabbing 2015, 1 for some reason, not the end of 2015 as instructed. From my debugging looks like the correct dates are supplied to Madrigal code but then the start of the year comes back for some unknown reason.

Also your example tries to load a day from 2017 instead of the day you asked to be downloaded.